src/components/Group/Group.js

```diff
@@ -5,85 +5,96 @@ import Task from '../Task/Task';
         group.tasks.map((i) => i.status ? ++notCompleted : ++completed);
-    }
-    
-    const addTask = (newTask) => {
-        group.tasks.push(newTask);
-        setGroups([...groups]);
-        setModalTaskVisible(false);
+        const updatedGroup = {
+            ...group,
+            completed,
+            notCompleted,
+          };
+        dispatch(fetchUpdateGroup({updatedGroup, groupId}));
+        setStatistics(updatedGroup);
     }
     
     const removeTask = (task) => {
-        let i = group.tasks.indexOf(task);
-        if(i >= 0) {
-            group.tasks.splice(i,1);
-        }
-        setGroups([...groups]);
+        const taskId = task._id
+        dispatch(fetchDeleteTask({groupId, taskId}));
     }
     
     const statusTask = (task) => {
-        group.tasks.filter( i => {
-            if (i.id === task.id) i.status = !i.status;
-            return i;
-          }
-        )
-        setGroups([...groups]);
+        const updatedTask = {
+            ...task,
+            status: !task.status
+          };
+        const taskId = task._id
+        dispatch(fetchUpdateTask({updatedTask, groupId, taskId}));
     }
 
     const editTask = (task) => {
-        setEdit(task.id);
+        setEdit(task._id);
         setValue(task.title);
     }
     
     const saveTask = (task) => {
-        group.tasks.filter( i => {
-            if (i.id === task.id) i.title = value;
-            setEdit(null);
-            setValue('');
-            return i;
-        }
-        )
-        setGroups([...groups]);
+        const updatedTask = {
+            ...task,
+            title: value
+          };
+        const taskId = task._id
+        setEdit(null);
+        setValue('');
+        dispatch(fetchUpdateTask({updatedTask, groupId, taskId}));
+    }
+
+    const priorityTask = (task) => {
+        const updatedTask = {
+            ...task,
+            priority: !task.priority
+          };
+        const taskId = task._id
+        dispatch(fetchUpdateTask({updatedTask, groupId, taskId}));
     }
     return (
         <div>
+            <div>
+                <Link className={styles.link} to='/statistics/'>
+                    <MyButton onClick={() => checkCompleted(group)} >Статистика групи</MyButton>
+                </Link>
+            </div>
             <h1 className={styles.groupHeader}>{group.title}</h1>
             <MyButton onClick={() => setModalTaskVisible(true)}>
                 Створити задачу
             </MyButton>
             <MyModal visible={modalTaskVisible} setVisible={setModalTaskVisible}>
-                <AddTaskForm add={addTask}/>
+                <AddTaskForm id={group._id} setVisible={setModalTaskVisible} />
             </MyModal>
             <MyButton onClick={() => removeGroup(group)}>
                   Видалити групу
             </MyButton>
             {group.tasks.length ? 
                 <div>
                 {group.tasks.map((task, index) => <Task 
```

Comment:
After destructuring `group` object:
```suggestion
            {tasks.length ? 
                <div>
                {tasks.map((task, index) => <Task 
```